netket/graph/grid.py

```diff
@@ -39,8 +39,53 @@ def __call__(self, sites):
     def __repr__(self):
         return f"T{self.shifts}"
 
-    def __hash__(self):
-        return hash((self.shifts, self.dims))
+
+@dataclass(frozen=True)
+class PlanarRotation(Element):
+
+    num_quarter_rots: int
+    axes: Tuple[int]
+    dims: Tuple[int]
+
+    def __call__(self, sites):
+        sites = sites.reshape(self.dims)
+        apply_perm = _np.arange(len(self.dims))
+        apply_perm[list(self.axes)] = self.axes[::-1]
+        for i in range(self.num_quarter_rots):
+            sites = sites.transpose(apply_perm)
+            sites = _np.flip(sites, self.axes[0])
+
+        return sites.ravel()
+
+    def __repr__(self):
+        if self.num_quarter_rots == 0:
+            return f"R(0,{self.axes})"
+        elif self.num_quarter_rots == 2:
+            return f"R(π,{self.axes})"
+        else:
+            return f"R({self.num_quarter_rots}π/2,{self.axes}"
```

Comment:
```suggestion
    def __repr__(self):
        return f"Rot({self.num_quarter_rots / 2:.1f}π, axes={self.axes})"
```
`Rot(0.5π, ...)`, `Rot(1.5π, ...)`, etc. all have the same width which can look be a bit more clean when printed.